Sources/Quick/TestState.swift

```diff
@@ -15,53 +15,56 @@ public struct TestState<T> {
     /// Resets the property to nil after each test.
     public init() {
         if AsyncWorld.sharedWorld.currentExampleGroup != nil {
-            AsyncWorld.sharedWorld.afterEach { [container] in
-                container.value = nil
+            AsyncWorld.sharedWorld.beforeEach { [container] in
+                TestStateConfiguration.afterEach.append {
```

Comment:
I wonder if this might be a good place to use [`XCTestCase.addTeardownBlock(_:)`](https://developer.apple.com/documentation/xctest/xctestcase/3815521-addteardownblock), as that should run when XCTest would invoke the `tearDown` method, which will necessarily be after all of the `afterEach` blocks run.

Reply:
Nice. I updated it to use `QuickSpec.current` and `AsyncSpec.current` to get the current test case. Let me know if that that's not the right way to do that, it's the only way  could find.